phoenicis-repository/src/main/java/org/phoenicis/repository/types/GitRepository.java

```diff
@@ -45,7 +45,10 @@
     private final LocalRepository.Factory localRepositoryFactory;
 
     private final File localFolder;
-    private static Semaphore mutex = new Semaphore(1);
+    // lock file to avoid concurrent access to the git clone
+    private final File lockFile;
+
+    private static Object mutex = 0;
```

Comment:
I think I would prefer:
```java
private static final Object mutex = new Object();
```
In my opinion `0` is normally used as a primitive value which can't be locked.